spec/units/frame_spec.rb

```diff
@@ -0,0 +1,221 @@
+require 'frame'
+
+describe Frame do
+
+  subject(:frame) { Frame.new(1) }
+
+  let(:roll_2) { double('roll', pins: 2) }
+  let(:roll_3) { double('roll', pins: 3) }
+  let(:roll_8) { double('roll', pins: 8) }
+  let(:roll_10) { double('roll', pins: 10) }
+
+
+  it 'has a frame number between 1-10' do
+    frame_1 = Frame.new(1)
+    expect(frame_1.number).to eq 1
+
+    frame_10 = Frame.new(10)
+    expect(frame_10.number).to eq 10
+  end
+
+  it 'throws error if frame_num < 1' do
+    expect{ Frame.new(0) }.to raise_error(RuntimeError)
+  end
+
+  it 'throws error if frame_num > 10' do
+    expect{ Frame.new(11) }.to raise_error(RuntimeError)
+  end
+
+  describe '#add_roll(roll)' do
+    it 'adds the roll to rolls if roll count < 2 AND new frame score would be <= 10' do
+      frame.add_roll(roll_2)
+      frame.add_roll(roll_8)
+      expect(frame.rolls).to include(roll_2)
+      expect(frame.rolls).to include(roll_8)
+    end
```

Comment:
For this test and the others below, it's useful to ask yourself - how does the user of this class use this class. I would bet that they would use `add_roll` to input things, and then use `score` to get the result. Is that is indeed the case, then these are the methods you should be using for testing. Not testing the values of actual instance variables, which would fix your implementation into something specific.

So something like:
```rb
    it 'adds the roll to rolls if roll count < 2 AND new frame score would be <= 10' do
      frame.add_roll(roll_2)
      frame.add_roll(roll_8)
      expect(frame.score).to eq 10
    end
```

Reply:
BTW, the test description seems odd.